crates/swc_ecma_transforms_module/src/system_js.rs

```diff
@@ -0,0 +1,1224 @@
+use anyhow::Context;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use swc_atoms::{js_word, JsWord};
+use swc_common::{BytePos, FileName, Mark, Span, SyntaxContext, DUMMY_SP};
+use swc_ecma_ast::*;
+use swc_ecma_utils::{quote_ident, quote_str, var::VarCollector, ExprFactory};
+use swc_ecma_visit::{noop_fold_type, Fold, FoldWith, VisitWith};
+
+use crate::{
+    path::{ImportResolver, NoopImportResolver},
+    util::{local_name_for_src, use_strict},
+};
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+#[serde(deny_unknown_fields, rename_all = "camelCase")]
+pub struct Config {
+    #[serde(default)]
+    pub allow_top_level_this: bool,
+}
+
+struct SystemJs<R>
+where
+    R: ImportResolver,
+{
+    root_mark: Mark,
+    resolver: Option<(R, FileName)>,
+    config: Config,
+
+    declare_var_idents: Vec<Ident>,
+    export_map: HashMap<(JsWord, Option<SyntaxContext>), Vec<JsWord>>,
+    export_names: Vec<JsWord>,
+    export_values: Vec<Expr>,
```

Comment:
I think this seems like a wrong design.

Reply:
Rust does **not** have GC

Reply:
Can you point at it? I'm not clear =.=

Reply:
Sorry, I thought this was used to store `clone()`-d values at the moment of reviewing this line. 

I forgot removing this comment after reviewing the code below.